distributed/shuffle/tests/test_rechunk.py

```diff
@@ -1275,3 +1271,43 @@ async def test_partial_rechunk_homogeneous_distribution(c, s, *workers):
     nchunks = [len(w.data.keys() & out_keys) for w in workers]
     # There are 40 output chunks and 5 workers. Expect exactly 8 chunks per worker.
     assert nchunks == [8, 8, 8, 8, 8]
+
+
+@gen_cluster(client=True, nthreads=[], config={"optimization.fuse.active": False})
+async def test_partial_rechunk_taskgroups(c, s):
+    arr = da.random.random((40, 40), chunks=(4, 4))
+    arr = arr.rechunk((1, -1), method="p2p")
+
+    _ = c.compute(arr)
+    await async_poll_for(
+        lambda: any(
+            isinstance(task, str) and task.startswith("shuffle-barrier")
+            for task in s.tasks
+        ),
+        timeout=5,
+    )
+    # Ensure that independent partial rechunks do not blow up the number of task groups
+    assert (
+        sum(
+            1
+            for task in s.task_groups
+            if isinstance(task, str) and task.startswith("rechunk-transfer")
+        )
+        == 1
+    )
+    assert (
+        sum(
+            1
+            for task in s.task_groups
+            if isinstance(task, str) and task.startswith("shuffle-barrier")
+        )
+        == 1
+    )
+    assert (
+        sum(
+            1
+            for task in s.task_groups
+            if isinstance(task, str) and task.startswith("rechunk-p2p")
+        )
```

Comment:
`sum(1 for ... if ...)` might be more readily understood as `len([x for x in L if ...])`

Just a nit though, please feel free to ignore.